### ISSUE TITLE
Add methods to UserProfilePersonalizationSettings for retrieving lock screen and desktop background

### ISSUE THREAD

**Issue by Jaiganeshkumaran** [microsoft/ProjectReunion#662]
# Proposal: Add methods to UserProfilePersonalizationSettings for retrieving lock screen and desktop backgrounds
Today UserProfilePersonalizationSettings has methods to set the lock screen and desktop background asynchronously. I understand that getting the current wallpaper could violate privacy however some customization apps may need to access the lock screen/desktop background to do their job.

## Summary
To methods TryGetLockScreenImageAsync and TryGetWallpaperImageAsync could be added to Windows.System.UserProfile.UserProfilePersonalizationSettings. Since this is privacy sensitive, Windows could show a dialog box asking the user whether the user wants to allow the app to access the lock screen/desktop background once. Instead of a permission or a capability, this can be asked every time the app calls the API.

## Rationale

* More opportunity for developers writing apps that target the app container

## Scope

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to access desktop/lockscreen background while being inside a low IL process | Must |
| This proposal will allow end users to let apps access their desktop/lockscreen background | Should |
| This proposal will allow developers to access desktop/lockscreen background silently | Won't |

## Important Notes
Same dialog should be shown when the API is accessed from unpackaged apps although those kinds of apps can use old Win32 APIs to achieve the same which won't show the consent dialog.

**Comment by Gavin-Williams** [comment]
There is no violation of privacy in a contract based security model. They simply need to add the alert to the application contract. And no dialog is required. Upon installation, the user agrees to the application's access to personalization data such as background image.

**Comment by jonwis** [comment]
Our pattern of declared capability (in the app manifest) and runtime use (the `Request...`) apis lets the user know that an app might ask for a power, and makes it clear when the app is asserting that power.  Adding the APIs you suggest is interesting, but it'd be behind a combination of those two.  The first time the app wants to use the power, the system asks the user - "is it OK for this app to change your background?" and there should be an "always" option so the user is never prompted again. (Although the app needs to `Request...` each time, subsequent requests may not actually pop the dialog if the user said "always.")

**Comment by jonwis** [comment]
@Jaiganeshkumaran , can you briefly describe a hypothetical app that would get the user's lockscreen/background images, and what the app would want to do with those?  You mention personalization, can you elaborate?

**Comment by riverar** [comment]
Theoretical app could provide additional features such as daily rotating imagery from a custom source, imagery that changes with day/night cycle, or even other triggers like ProjectReunion build failures.

Theoretical app could appear as such:
![image](https://user-images.githubusercontent.com/475132/113398504-910c4080-9353-11eb-9ed0-d1c3cbc45c31.png)

**Comment by ahmed605** [comment]
An example of what this feature or capability can be used for: acrylic wallpaper tinting like in MacOS
So you would request the wallpaper -> get the dominant color -> cache it -> set it as the acrylic tint color

**Comment by tbolon** [comment]
See also windows [Dynamic Theme](https://www.microsoft.com/en-us/p/th%C3%A8me-dynamique/9nblggh1zbkw?source=lp&activetab=pivot:overviewtab) store app.

**Comment by FireCubeStudios** [comment]
We need this now

**Comment by FireCubeStudios** [comment]
I request this functionality be added with Windows 10 support too

**Comment by FireCubeStudios** [comment]
> @Jaiganeshkumaran , can you briefly describe a hypothetical app that would get the user's lockscreen/background images, and what the app would want to do with those? You mention personalization, can you elaborate?

I would use to get wallpaper colors to achieve similair things to this https://twitter.com/LunaNeither/status/1397603160567468042

**Comment by Joshagos** [comment]
Why not simply just use Winaero Tweaker to extract the images?

**Comment by Jaiganeshkumaran** [comment]
> Why not simply just use Winaero Tweaker to extract the images?
You can't do that programmatically in your UWP app. Also, Winaero Tweaker is a Win32 desktop app that reads from the registry which an UWP app can't do that easily.

**Comment by mominshaikhdevs** [comment]
Part Of: 

* #219.

**Comment by mominshaikhdevs** [comment]
Part Of: 

* #8.

* #219.

**Comment by riverar** [comment]
@mominshaikhdevs Do you work at Microsoft? Why are you adding these comments to various issues in the repository?

**Comment by castorix** [comment]
BTW, there is [LockScreen Class](https://learn.microsoft.com/en-us/uwp/api/windows.system.userprofile.lockscreen?view=winrt-22621) and [IDesktopWallpaper interface](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-idesktopwallpaper)

**Comment by riverar** [comment]
@mominshaikhdevs It wasn't obvious to me or I wouldn't have asked. I would appreciate if you stop adding a bunch of noise to the issues in this repository.

**Comment by mominshaikhdevs** [comment]
> [@mominshaikhdevs](https://github.com/mominshaikhdevs) ...a bunch of noise...

I disagree with your perception of "a bunch of noise". I would appreciate too if you do the reverse.
